base_layer/wallet/src/output_manager_service/service.rs

```diff
@@ -617,14 +695,19 @@ where
     }
 
     async fn create_output_with_features(
-        &self,
+        &mut self,
         value: MicroTari,
         features: OutputFeatures,
     ) -> Result<UnblindedOutputBuilder, OutputManagerError> {
-        let (spending_key, script_private_key) = self
+        let spending_key = self
             .resources
             .master_key_manager
-            .get_next_spend_and_script_key()
+            .get_next_key(KeyManagerOmsBranch::Spend.to_string())
```

Comment:
So this operation was done together before because these values need to use the same index. What happens if these two indices get out of sync somehow?

Reply:
furthermore, you have undone a huge amount of DRYing up by turning this into two calls here in the client service.